advanced_survey/views.py

```diff
@@ -61,10 +61,32 @@ def save_survey(request): # pylint: disable=R0912
                     if 'description' in question:
                         db_question.description = question['description']
                     db_question.field_type = question['field']
-                    if 'url' in question:
-                        db_question.options_url = question['url']
                     if 'option' in question:
-                        db_question.options = question['option']
+                        options = question['option']
+                        if 'url' in options:
+                            if 'jsonpath' not in options or options['jsonpath'] == "":
+                                raise RuntimeError(
+                                    f"json path for url {options['url']} is required"
+                                )
+                            request = urllib.request.urlopen(options['url'])
+                            if request.status != 200:
+                                raise RuntimeError(
+                                    f"get request for {options['url']} is {request.status}"
+                                )
+                            response = json.loads(request.read())
+                            if isinstance(response, dict):
+                                fields = options['jsonpath'].split(".")
+                                values = []
+                                for field in fields:
+                                    if field not in response:
+                                        raise RuntimeError(f"field {field} not found")
+                                    values = response[field]
```

Comment:
Not sure I understand this. If `fields` is `foo.bar.waldo`, we check if `response` has properties `foo`, `bar`, and `waldo`?

Are we supposed to do `values += response[field]`? Something doesn't make sense here

Reply:
Good catch, Fixed. We don't need `values`.